hikari/files.py

```diff
@@ -733,17 +737,21 @@ class URL(WebResource):
         a `builtins.bytes` and pass that instead in these cases.
     """
 
-    __slots__: typing.Sequence[str] = ("_url",)
+    __slots__: typing.Sequence[str] = ("_url", "_filename")
 
-    def __init__(self, url: str) -> None:
+    def __init__(self, url: str, filename: typing.Optional[str] = None) -> None:
         self._url = url
+        self._filename = filename
 
     @property
     def url(self) -> str:
         return self._url
 
     @property
     def filename(self) -> str:
+        if self._filename:
+            return self._filename
```

Comment:
test case to make sure this doesn't ever get accidentally reverted?

Reply:
I might have forgotten to push it. Will do so later!

Reply:
Added!